server/models/service-instance.js

```diff
@@ -0,0 +1,70 @@
+var async = require('async');
+var debug = require('debug')('MeshModels.server.ServiceInstance');
+
+module.exports = function(ServiceInstance) {
+  ServiceInstance.beforeRemote(
+    'prototype.updateAttributes',
+    function(ctx, _, next) {
+      debug('updateAttributes with %j', ctx.args.data);
+      // When updating the instance via REST, only allow changes to cpus
+      ctx.args.data = {
+        cpus: ctx.args.data.cpus
+      };
+      next();
+    }
+  );
+
+  ServiceInstance.observe('before save', function beforeUpdate(ctx, next) {
+    if (ctx.instance) {
+      // create or full update of instance model
+      if (isNaN(parseInt(ctx.instance.cpus, 10))) {
+        ctx.instance.cpus = 'CPU';
+      }
+    } else if (ctx.data && ctx.data.cpus &&
+      isNaN(parseInt(ctx.data.cpus, 10))) {
+      ctx.data.cpus = 'CPU';
+    }
+    next();
+  });
+
+  ServiceInstance.observe('after save', function(ctx, next) {
```

Comment:
This code is pretty much cut-n-paste, and verbose: I see a copy in at least server-service.js, I think it would be better refactored into some common place. That would also allow it to be clear that the code is actually the same... peering closely, I see some minor differences (this version is shorter by pulling serviceManager out as a var), but I think its otherwise the same.